src/client.rs

```diff
@@ -236,18 +237,40 @@ impl Client {
 
         add_invite_event_handler(client, tx, &user_id).await;
         add_room_message_event_handler(client, tx, &user_id, &self.event_notifier).await;
+        add_created_room_event_handler(client, user_notifier).await;
 
         let (cancel_sync, check_cancel) = async_channel::bounded::<bool>(1);
 
         tokio::spawn(sync_until_cancel(client, check_cancel).await);
 
         let res = response.expect("already checked it is not an error");
-        let joined_rooms = res.rooms.join.keys().cloned().collect::<Vec<_>>();
+        let mut joined_rooms = res.rooms.join.keys().cloned().collect::<Vec<_>>();
         let invited_rooms = res.rooms.invite.keys().cloned().collect::<Vec<_>>();
+        let mut channels: Vec<OwnedRoomId> = Vec::new();
+        let mut indexes: Vec<usize> = Vec::new(); // collect indexes to remove from joined_rooms
+
+        for (i, id) in joined_rooms.iter().enumerate() {
+            match client.get_room(id) {
+                Some(room) => {
+                    if !room.is_direct() && room.is_public() {
+                        channels.push(id.to_owned());
+                        indexes.push(i);
+                    }
+                },
+                None => log::debug!("room not found in store {}", id)
+            }
+        }
+
+        // remove public rooms from joined room
+        for i in indexes {
+            joined_rooms.remove(i);
```

Comment:
This might be a good time to rename these to `direct_messages` as the `room` concept applies to both channels and DMs.